lib/phoenix_live_view/html_formatter.ex

```diff
@@ -391,7 +393,17 @@ defmodule Phoenix.LiveView.HTMLFormatter do
   end
 
   defp to_tree([{:tag_close, name, _meta} | tokens], buffer, [{name, attrs, upper_buffer} | stack]) do
-    tag_block = {:tag_block, name, attrs, Enum.reverse(buffer)}
+    format? =
+      case List.first(upper_buffer) do
+        {:text, text, _meta} ->
+          :binary.last(text) in '\s'
```

Comment:
Should this also consider tabs and newlines? If you have any of them, you want to format, right? Or are newlines considered elsewhere? In any case, it probably doesn't hurt to consider them here too.

```suggestion
          :binary.last(text) in '\s\t\r\n'
```